irksome/wso_dirk_tableaux.py

```diff
@@ -0,0 +1,183 @@
+      1.213637180023516e-01, -4.752289775376601e-01, 5.285605969257756e-01,
+      0, 0, 0, 0, 0, 0],
+     [3.391631788320480e-01, -2.797427027028997e-01, 1.039483063369094e+00,
+      5.978770926212172e-02, -2.132900327070380e-01, 8.344318363436753e-02,
+      2.410106515779412e-01, 0, 0, 0, 0, 0],
+     [5.904282488642163e+00, 3.171195765985073e+00, -1.236822836316587e+01,
+      -4.989519066913001e-01, 2.160529620826442e+00, 1.916104322021480e+00,
+      1.988059486291180e+00, 2.232092386922440e-01, 0, 0, 0, 0],
+     [4.616443509508975e-01, -1.933433560549238e-01, -1.212541486279519e-01,
+      6.662362039716674e-02, 4.254912950625259e-01, 7.856131647013712e-01,
+      8.369551389357689e-01, 1.604780447895926e-01, 3.616125951766939e-01,
+      0, 0, 0],
+     [-7.087669749878204e-01, 6.466527094491541e-01, 4.758821526542215e-01,
+      -2.570518451375722e-01, 1.123185062554392e+00, 5.546921612875290e-01,
+      3.192424333237050e-01, 3.612077612576969e-01, 5.866779836068974e-01,
+      2.353799736246102e-01, 0, 0],
+     [4.264162484855930e-01, 1.322816663477840e+00, 4.245673729758231e-01,
+      -2.530402764527700e+00, -7.822016897497742e-02, 1.054463080605071e+00,
+      4.645590541391895e-01, 1.145097379521439e+00, 4.301337846893282e-01,
+      1.499513057076809e+00, 1.447942640822165e-02, 0],
+     [1.207394392845339e-02, 5.187080074649261e-01, 1.121304244847239e-01,
+      -4.959806334780896e-03, -1.345031364651444e+00, 3.398828703760807e-01,
+      8.159251531671077e-01, -2.640104266439604e-03, 1.439060901763520e-02,
+      -6.556567796749947e-03, 6.548135446843367e-04, 5.454220210658036e-01]])
+
+wso1255c = numpy.array(
+    [4.113473525867655e-02, 2.269850660400232e-01, 6.222969192243949e-01,
+     1.377989449231234e+00, 1.259841986970257e+00, 1.228350442796143e+00,
+     1.269855051265635e+00, 2.496200652601413e+00, 2.783820705331141e+00,
+     3.337101417632813e+00, 4.173423133876636e+00, 1.000000000000000e+00])
+
+wso1255b = numpy.array(
+    [1.207394392845339e-02, 5.187080074649261e-01, 1.121304244847239e-01,
+     -4.959806334780896e-03, -1.345031364651444e+00, 3.398828703760807e-01,
+     8.159251531671077e-01, -2.640104266439604e-03, 1.439060901763520e-02,
+     -6.556567796749947e-03, 6.548135446843367e-04, 5.454220210658036e-01])
+
+wsodict = {
+    (4, 3, 2): (wso432A, wso432b, wso432c),
+    (4, 3, 3): (wso433A, wso433b, wso433c),
+    (6, 4, 3): (wso643A, wso643b, wso643c),
+    (7, 4, 4): (wso744A, wso744b, wso744c),
+    (12, 5, 4): (wso1254A, wso1254b, wso1254c),
+    (12, 5, 5): (wso1255A, wso1255b, wso1255c)
+}
+
+
+class WSODIRK(ButcherTableau):
+    def __init__(self, ns, order, ws_order):
+        try:
+            A, b, c = wsodict[ns, order, ws_order]
+        except KeyError:
+            raise NotImplementedError("No WSO DIRK method for that combination of stages, order and weak stage order")
+        self.ws_order = ws_order
+        super(WSODIRK, self).__init__(A, b, None, c, order)
```

Comment:
This will need updating if we merge the adaptive PR in its current form, to pass `None` for `Atilde` and `btilde`

Reply:
Good catch, we'll update it on the adaptive PR end after I get that reviewed.